std/algorithm/iteration.d

```diff
@@ -1524,14 +1540,26 @@ private struct GroupByImpl(alias pred, Range)
     }
 
     @property bool empty() { return impl.current.empty; }
-    @property auto front() { return Group(impl); }
+
+    @property auto front()
+    {
+        static if (isUnary)
+        {
+            import std.typecons : tuple;
+            return tuple(unaryFun!pred(impl.current.front), Group(impl));
```

Comment:
Hmm. To me the optimization seems to be the hard part. I think you will need to make assumptions about the key type, since you need to store it in some way. I am guessing here. In any case, the whole change is kind of pointless without the optimization. Thus, I would like to see it in before 2.067, if at all feasible.